### PR TITLE
[Follow-up] Make `RunningAverage` and `Rouge` serializable

### PR DESCRIPTION
**Pull request by sadra-barikbin**
Now `state_dict` could serialize metric attributes that themselves are metric or simple collections of metrics.(For `Rouge`). If an attribute is `None`, `state_dict` sets the same in the dict (for `_value` in `RunningAverage`)

### CODE REVIEW

File: ignite/metrics/metric.py [hunk 1]
```diff
@@ -554,32 +554,41 @@ def state_dict(self) -> OrderedDict:
...
+                    state[attr_name] = [attr]
+                else:
+                    if isinstance(attr, (int, float)):
+                        attr_type = type(attr)
+                        attr = float(attr)
+                    gathered_attr = idist.all_gather(attr)
+                    if isinstance(attr, float):
+                        gathered_attr = [attr_type(process_attr) for process_attr in cast(torch.Tensor, gathered_attr)]
+                    state[attr_name] = cast(Union[torch.Tensor, List], gathered_attr)
+            elif attr is None:
```
Comment (vfdev-5) [inline ignite/metrics/metric.py:584]: What is the usage for None ?

**Review by vfdev-5 — APPROVED** [review]
Thanks @sadra-barikbin, PR looks good to me!
Can you please check and report here how it works on MNIST while saving and resuming.
As a test example, please use: 
https://github.com/pytorch/ignite/blob/11a1fba0594f6ac0db9e8bc5f21305dc338bb7c1/examples/mnist/mnist_save_resume_engine.py
Here are few instructions on https://github.com/pytorch/ignite/blob/master/examples/mnist/README.md#training-save--resume
Finally, we can also update `mnist_save_resume_engine.py` code adding RunningAverage metric for batch loss to the trainer and plotting it.

**Comment by sadra-barikbin** [comment]
Here is the running loss until stop and after resume:
image

image

**Comment by sadra-barikbin** [comment]
Regarding your [comment](https://github.com/pytorch/ignite/pull/3001#discussion_r1289662406) in the main PR, I introduced a new if branch in `Checkpoint` to accept a dictionary of serializable objects as an item in the `to_save`. This way we could do
```python
metrics = {'acc': Accuracy(), 'nll': Loss()}
to_save={'model':model, 'trainer':trainer, 'metrics': metrics}
```
How is this?
